### PR TITLE
Remove iOS 11 support.

### PR DESCRIPTION
**Pull request by MaximAlien**
PR increases minimum deployment targets to:
- iOS 12.0
- tvOS 12.0
- macOS 10.14
- watchOS 5.0

### CODE REVIEW

File: Cartfile [hunk 1]
```diff
@@ -1,2 +1,2 @@
-github "raphaelmor/Polyline" ~> 5.0
+github "MaximAlien/Polyline" "master"
```
Comment (MaximAlien) [inline Cartfile:1]: New release of Polyline with updated deployment targets is needed. PR can be found here: https://github.com/raphaelmor/Polyline/pull/78.

File: MapboxDirections.xcodeproj/xcshareddata/xcschemes/MapboxDirections watchOS.xcscheme [hunk 2]
```diff
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <Scheme
-   LastUpgradeVersion = "1340"
+   LastUpgradeVersion = "1400"
```
Comment (1ec5) [inline MapboxDirections.xcodeproj/xcshareddata/xcschemes/MapboxDirections watchOS.xcscheme:3]: #730 will consolidate the schemes and targets, making future upgrades cleaner.

**Review by 1ec5 — APPROVED** [review]
Suggested changelog entry:

* This library now requires a minimum deployment target of iOS 12.0 or above, macOS 10.14.0 or above, tvOS 12.0 or above, or watchOS 5.0 or above. Older operating system versions are no longer supported. ([#736](https://github.com/mapbox/mapbox-directions-swift/pull/736))

**Review by 1ec5 — COMMENTED** [review]
Also update the podspec:

https://github.com/mapbox/mapbox-directions-swift/blob/8c275c9414f0a2be2e430fe11e24078c6c0498c1/MapboxDirections.podspec#L28-L31

File: Cartfile [hunk 3]
```diff
@@ -1,2 +1,2 @@
-github "raphaelmor/Polyline" == 5.0.3
+github "raphaelmor/Polyline" ~> 5.0
```
Comment (MaximAlien) [inline Cartfile:1]: There is no need to pin to specific version of  Polyline anymore.

**Pull request merged**